File: micro-services/bil-generator-service/routes/bilRoutes.py
from flask import Blueprint
from controllers.bilController import visualizar_boleto, download_boleto

bil_routes = Blueprint("bil_routes", __name__)

bil_routes.get("/boleto/<invoice_number>/visualizar")(visualizar_boleto)
bil_routes.get("/boleto/<invoice_number>/download")(download_boleto)

File: backend/routes/bil.routes.js
const express = require("express");
const router = express.Router();
const bilController = require("../controllers/bil.controller");

// Visualizar o boleto no navegador
router.get("/boleto/:invoiceNumber/visualizar", bilController.visualizarBoleto);

// Baixar o boleto como PDF
router.get("/boleto/:invoiceNumber/download", bilController.downloadBoleto);

module.exports = router;
